src/styles/main.scss

```diff
@@ -0,0 +1,51 @@
+$days: 31;
+$month-length: (28, 29, 30, 31);
+$start-day: (mon, thu, wed, thu, fri, sat, sun);
+$gap: 1px;
+$cell-size: 100px;
+
+body {
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  min-height: 100vh;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+@for $i from 1 through $days {
+  .calendar__day:nth-child(#{$i})::before {
+    content: '#{$i}';
+  }
+}
+
+.calendar {
+  gap: $gap;
+  display: flex;
+  flex-wrap: wrap;
+  width: 7 * ($cell-size + $gap) - $gap;
+
+  &__day {
+    margin: 0;
+    box-sizing: border-box;
+    height: $cell-size;
+    width: $cell-size;
+    background-color: #eee;
+    border: 1px solid #000;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    transition-duration: 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translate(0, -20px);
+    }
+  }
+}
+
+.calendar__day:first-child {
+  margin-left: 606px;
```

Comment:
The `margin-left` value of `606px` for `.calendar__day:first-child` is hardcoded, which is not a flexible solution. Consider implementing a more dynamic approach that calculates the offset based on the `$start-day` variable to accommodate different starting days of the month.